### PR TITLE
Add support for SIG OpenGraph tags

### PR DESCRIPTION
**Pull request by gmjuhasz**
This change makes the plugin use the SIG image for the OpenGraph tags, if it is enabled. Right know it's handled in two different places, but I used a common hook for the logic.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

1203876111187910-as-1203968442376755

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Before the new Social Image Generator service is merged, it can be only tested by abtracting that.

In `projects/packages/publicize/src/class-publicize-base.php` comment out `$sig       = new Social_Image_Generator();` in the 1499. line, and at it's usage below just provide a dummy string.

Do the following with Social standalone, then Jetpack:
* Create a new post and add a featured image. In the SIG settings disable SIG and publish the post ( no need to share it ). Open the created blogpost, and in the inspector search for the `og:image` tag. It should point to the featured image.
* Go back to the editor and enable SIG and update the post.
* Refresh the opened blogpost, now the `og:image` should contain the dummy string you specified before

### CODE REVIEW

File: projects/packages/publicize/src/class-publicize-base.php
Comment (danielpost) [inline projects/packages/publicize/src/class-publicize-base.php]: I've [added a method for checking the status to the SIG library](https://github.com/Automattic/jetpack/pull/29118) so we'll be able to use that instead.

As an aside, it's always good to check that a variable exists and has the same type as you're expecting before using it. In this case, if `$social_options` wasn't an array or didn't have the `image_generator_settings` key, this would throw a warning.

File: projects/packages/publicize/src/class-publicize-base.php
Comment (danielpost) [inline projects/packages/publicize/src/class-publicize-base.php]: I've [slightly changed this method in the library](https://github.com/Automattic/jetpack/pull/29118) so we'll need to update this, but I think it's probably best to hold off on updating this until that PR is merged 👍🏼

File: projects/plugins/social/src/class-meta-tags.php
Comment (danielpost) [inline projects/plugins/social/src/class-meta-tags.php]: I'm not 100% sure if we should use this filter to add the image. It runs on `Jetpack_PostImages::get_image()`, which is also used by other things (such as Related Posts).

What do you think about hooking into the `jetpack_open_graph_tags` filter instead? We could try to add it to Social the same way you did here.

File: projects/packages/publicize/src/class-publicize-base.php
Comment (gmjuhasz) [reply projects/packages/publicize/src/class-publicize-base.php]: Noted!

File: projects/packages/publicize/src/class-publicize-base.php
Comment (gmjuhasz) [inline projects/packages/publicize/src/class-publicize-base.php]: TBD: Update this when lib gets merged

File: projects/plugins/jetpack/functions.opengraph.php
Comment (danielpost) [inline projects/plugins/jetpack/functions.opengraph.php]: Do we need to add this? Would it be possible to just use `get_post()` or `get_the_ID()` in the filter instead?

Comment (gmjuhasz) [reply projects/plugins/jetpack/functions.opengraph.php]: My core WP knowledge failed me! Good idea.

File: projects/packages/publicize/src/class-publicize-base.php
Comment (gmjuhasz) [reply projects/packages/publicize/src/class-publicize-base.php]: Done

File: projects/packages/publicize/src/class-publicize-base.php
Comment (danielpost) [inline projects/packages/publicize/src/class-publicize-base.php]: We need to use the utility function instead: https://github.com/Automattic/jetpack/blob/trunk/projects/packages/publicize/src/social-image-generator/utilities.php#L18

The utility function returns the actual generated URL. The `Post_Settings()->get_image_url()` just returns the background image for the generated image.

**Review by danielpost — CHANGES_REQUESTED** [review]

**Comment by danielpost** [comment]
I just tested this and it adds the image URL correctly. However, it appears that the URL gets escaped in a way that turns `&` into `&amp;`, which breaks the URL. It's a bug in the `render_tags` method in the Jetpack Social meta tags class (I didn't test with Jetpack).

I think adding something like `str_replace( '&amp;', '&', $url );` should fix it.

**Review by danielpost — APPROVED** [review]

**Pull request merged**
